app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisController.kt

```diff
@@ -0,0 +1,154 @@
+import co.netguru.baby.monitor.client.common.base.ServiceController
+import co.netguru.baby.monitor.client.data.DataRepository
+import co.netguru.baby.monitor.client.feature.babynotification.NotifyBabyEventUseCase
+import co.netguru.baby.monitor.client.feature.debug.DebugModule
+import co.netguru.baby.monitor.client.feature.machinelearning.MachineLearning
+import co.netguru.baby.monitor.client.feature.noisedetection.NoiseDetector
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.rxkotlin.subscribeBy
+import io.reactivex.schedulers.Schedulers
+import timber.log.Timber
+import javax.inject.Inject
+
+class VoiceAnalysisController @Inject constructor(
+    private val notifyBabyEventUseCase: NotifyBabyEventUseCase,
+    private val sharedPrefsWrapper: FirebaseSharedPreferencesWrapper,
+    private val notificationHandler: NotificationHandler,
+    private val debugModule: DebugModule,
+    private val noiseDetector: NoiseDetector,
+    private val dataRepository: DataRepository,
+    private val machineLearning: MachineLearning
+) : ServiceController<VoiceAnalysisService> {
+
+    private val compositeDisposable = CompositeDisposable()
+    private var aacRecorder: AacRecorder? = null
+    private var voiceAnalysisService: VoiceAnalysisService? = null
+    var voiceAnalysisOption =
+        VoiceAnalysisOption.MachineLearning
+        set(value) {
+            field = value
+            aacRecorder?.voiceAnalysisOption = value
+        }
+
+    override fun attachService(service: VoiceAnalysisService) {
+        this.voiceAnalysisService = service
+        notificationHandler.showForegroundNotification(service)
+        dataRepository.getClientData()
+            .subscribeOn(Schedulers.io())
+            .subscribeBy(onSuccess = {
+                voiceAnalysisOption = it.voiceAnalysisOption
+                startRecording()
+            }, onComplete = this::startRecording)
+            .addTo(compositeDisposable)
+    }
+
+    fun startRecording() {
+        if (aacRecorder != null) return
+        initBabyEventsSubscription()
+        aacRecorder =
+            AacRecorder(voiceAnalysisOption)
+                .apply {
+                    startRecording()
+                        .subscribeOn(Schedulers.io())
+                        .subscribeBy(
+                            onComplete = { Timber.i("Recording completed") },
+                            onError = Timber::e
+                        ).addTo(compositeDisposable)
+                    machineLearningData
+                        .filter { voiceAnalysisOption == VoiceAnalysisOption.MachineLearning }
+                        .subscribeOn(Schedulers.newThread())
+                        .subscribeBy(
+                            onNext = this@VoiceAnalysisController::handleRecordingData,
+                            onComplete = { Timber.i("Complete") },
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/89ff9ffb603a83b79e177dd00519720c6a8d52ac/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisController.kt#L68"></span>Multiple occurrences of the same string literal within a single file detected.</td>
    </tr>
  </tbody>
</table>